File: Desktop/DSA/T73/maximumsubarraywithzerosum.py
# Problem statement
# Given an array arr of length N consisting of positive and negative integers, return the length of the longest subarray whose sum is zero.

def maximum_subarray(array:list[int])->int:
    
    n=len(array)
    max_len=float("-inf")
    
    for start in range(n):
        
        curr = 0
        
        for end in range(start,n):
            
            curr +=array[end]
            
            if curr == 0:
                
                length = end-start+1
                
                max_len = max(max_len,length)
                
    return max_len

arr = [1, 2, -3, 3, 1, -1, 2, -2]
print(maximum_subarray(arr))

File: Documents/ANUDIP JAVA/Maximumsubarraywithzerosum.java


public class Maximumsubarraywithzerosum {
	
	public static int zerosum(int[] nums) {
		
		int maxlength = 0;
		
		for (int i=0 ;i<nums.length ; i++) {
			
			int sum = 0;
			
			for (int j= i; j <nums.length ;j ++) {
				
				sum += nums[j];
				
				if (sum == 0) {
					
					maxlength = Math.max(maxlength, j-i+1);
				}
			}
		}
		
		return maxlength;
	}
	
	public static void main(String[]args) {
		
		 int[] arr = {1, 2, -3, 3, 1, -4, 2, 1, -1, -2};

	        int maxLength = zerosum(arr);

	        System.out.println("The length of the longest subarray with zero sum is: " + maxLength);
	}
}
